app/src/main/java/com/example/sharingang/MainActivity.kt

```diff
@@ -1,7 +1,49 @@
 package com.example.sharingang
 
+import android.content.pm.PackageManager
+import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts.RequestPermission
 import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
-class MainActivity : AppCompatActivity(R.layout.activity_main)
\ No newline at end of file
+class MainActivity : AppCompatActivity(R.layout.activity_main) {
+
+    var permissionGranted: Boolean = false
+
+    private val requestPermissionLauncher =
+        registerForActivityResult(RequestPermission()) { isGranted: Boolean ->
+            permissionGranted = if (isGranted) {
+                true
+            } else {
+                Toast.makeText(
+                    this,
+                    "Storage permission successfully denied. Feature is disabled.",
+                    Toast.LENGTH_LONG
+                ).show()
+                false
```

Comment:
You could probably write this segment as like:

```kotlin
if (!isGranted) {
  doSomething()
}
isGranted
```

Reply:
You're right, I'll change that.